### PR TITLE
Port su program from NetBSD

### PR DESCRIPTION
**Pull request by franciscozdo**
I removed some code from BSD implementation (macro `BSD4_4`). I don't think we will use it.

Added syscalls which do nothing but `su` needs them.
* `getpriority()`
* `setpriority()`

Added some dummy functions that do nothing just return 0 or "" to avoid commenting out code in su. 
* `getusershell()`
* `setusershell()`
* `ttyname()`
* `crypt()`
* `getpass()`

### CODE REVIEW

File: usr.bin/su/su.c [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+	enum { UNSET, YES, NO } iscsh = UNSET;
+	const char *user, *shell, *avshell;
+	char *username, **np;
+#ifdef SU_ROOTAUTH
+	char *userpass;
+#endif
+	char *class;
+	char shellbuf[MAXPATHLEN], avshellbuf[MAXPATHLEN];
+#ifdef LOGIN_CAP
+	time_t pw_warntime = _PASSWORD_WARNDAYS * SECSPERDAY;
```
Comment (franciscozdo) [inline usr.bin/su/su.c:105]: This line was moved here (between macro) because it is only used when `LOGIN_CAP` is used.

**Review by pj1031999 — APPROVED** [review]
LGTM

**Comment by cahirwpz** [comment]
Could we adapt:
 * `getusershell` & `setusershell` from http://bxr.su/OpenBSD/lib/libc/gen/getusershell.c
 * `ttyname` from http://bxr.su/OpenBSD/lib/libc/gen/ttyname.c (let's just comment out lines 90 - 107)
?

**Comment by franciscozdo** [comment]
> Could we adapt:
> 
> * `getusershell` & `setusershell` from http://bxr.su/OpenBSD/lib/libc/gen/getusershell.c
> * `ttyname` from http://bxr.su/OpenBSD/lib/libc/gen/ttyname.c (let's just comment out lines 90 - 107)
>   ?

I have adapted `getusershell` and `setusershell` but `ttyname` left. IMO there is too much to add. It must be done better than just copying from BSD.

**Pull request merged**